File: Python/Data Structures & Algorithms Module - Python/Week1/Fibonacci.py
"""
Fibonacci.py
Billy Cussen
03/11/2020
"""

def fibonacci(num):
    a, b, c = 0, 1, 0
    print("Number is "+str(a)+" at Position 1")
    print("Number is "+str(b)+" at Position 2")
    i = 2
    while i <= num:
        c = a + b
        print("Number is "+str(c)+" at Position "+str(i))
        a = b
        b = c
        i = i + 1

def fibRecursion(num):
    if num == 0 or num == 1:
        return 1
    return fibRecursion(num-1)+fibRecursion(num-2)

print("***RECURSION LOOPED APPROACH***")
fibonacci(10)

print("***RECURSION RECURSIVE APPROACH***")
for i in range(10):
    print("Number is "+str(fibRecursion(i))+" at Position "+str(i))

File: Java/Data Structures & Algorithms Module - Java/Week8/Revision/Week7/Fibonacci.java
/*
Fibonacci.java
Billy Cussen
17/11/2020
*/
public class Fibonacci{

    public static void main(String [] args){

        for(int i = 0; i < 10; i++){
            System.out.println("Fib Looped: "+fib(i)+", Fib Recursion: "+fibRecursion(i));
        }
    }

    public static int fib(int n){
        int a = 0, b = 1, c = 1;
        for(int i = 0; i < n; i++){
            c = a + b;
            a = b;
            b = c;
        }
        return c;
    }

    public static int fibRecursion(int n){
        if(n == 0 || n == 1){
            return 1;
        }
        return fibRecursion(n-1)+fibRecursion(n-2);
    }
}